TROUBLESHOOTING.md

```diff
@@ -36,7 +36,45 @@ to index for features such as go to definition.
 
 ## Common issues
 
-There are two main sources of issues users typically face during activation: shell or Bundler related problems.
+There are several main sources of issues users typically face during activation: outdated version, shell problems, or Bundler related problems.
+
+### Outdated Version
+
+> [!NOTE] If using VS Code, the version of the extension is distinct from that that of the server (the `ruby-lsp` gem). You can check the server version in the Output panel upon startup.
+
+In most cases, the server gem will be automatically updated. You can also trigger a manual update with the "Update language server gem" command in VS Code.
+
+You can also attempt an update from the command line with `BUNDLE_GEMFILE=.ruby-lsp/Gemfile bundle update ruby-lsp`
+
+> [!NOTE] If you're using any addon gem, such as `ruby-lsp-rspec`, then `ruby-lsp` will also be present in your `Gemfile.lock` and will not receive automatic updates. You will need to keep `ruby-lsp` up-to-date as you would with any other gem, e.g. using Dependabot.
```

Comment:
This isn't fully correct. Even if the `ruby-lsp` is locked, we will still update it. The only time when it won't get updated is if another gem is restricting the version somehow.

Reply:
Updated